src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -102,6 +102,46 @@ internal class AssemblyLoader
             return discoveredAssemblyLocations;
         }
 
+        public static T TryLoadAndCreateInstance<T>(string assemblyName, TraceLogger logger) where T : class
+        {
+            try
+            {
+                var exeRoot = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
```

Comment:
As I learned investigating https://github.com/dotnet/orleans/issues/593, this method of loading referenced assemblies doesn't seem to work in ASP.NET. See the fix in https://github.com/dotnet/orleans/pull/594 I had to make to make it load assemblies.